server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -139,8 +140,10 @@ public static List<DfsKnnResults> mergeKnnResults(SearchRequest request, List<Df
         }
 
         List<List<TopDocs>> topDocsLists = new ArrayList<>(request.source().knnSearch().size());
+        List<SetOnce<String>> nestedPath = new ArrayList<>(request.source().knnSearch().size());
```

Comment:
What's the reason we need SetOnce here, why not just List<String> ?

Reply:
Honestly, I am thinking we should have an assertion with the `setonce`. We should ONLY see a single nested path (as we are querying the same knn field over all the indices/shards). Seeing more than one nested path for the same knn clause declaration would be bad.